### PR TITLE
MueLu: update to the User Guide

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/muelu 

## Description
Adding new options in the user guide and modifying accordingly some factories to expose the new options. Also modifying some xml files in trilinosCouplings once again to reflect the newly available options.

## Motivation and Context
The MueLu User Guide has not been updated in a while and a significant amount of changes have occurred since then so it's time to get this done.

## How Has This Been Tested?
Well, it's the documentation so it's not tested...

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.

### CODE REVIEW

**Comment by lucbv** [comment]
At this time it's a work in progress so please keep the `AT: WIP` label on until we all agree that the new User Guide is looking good!

**Comment by cgcgcg** [comment]
@jhux2 `latexdiff` does that..

**Comment by jhux2** [comment]
> `latexdiff` does that..

I'll have a look, thanks!

**Comment by csiefer2** [comment]
Structured stuff looks good to me.

**Comment by cgcgcg** [comment]
Pushed some changes

File: packages/muelu/doc/UsersGuide/SANDdistribution.tex [hunk 1]
```diff
@@ -5,45 +5,39 @@
...
-    \SANDdistInternal{1}{1322}{Christopher Siefert}{1443}
+{\bfseries DISTRIBUTION}
 
-\end{SANDdistribution}
+{\bfseries Email--External (encrypt for OUO)}\\[1ex]
+\begin{tabular}{|p{5cm}|p{4cm}|p{4cm}|}
+  \hline
+  \multicolumn{1}{|p{5cm}}{\cellcolor{SANDgreen}\bfseries Name} & \multicolumn{1}{|p{4cm}}{\cellcolor{SANDgreen}\bfseries Company Email Address} & \multicolumn{1}{|p{4cm}|}{\cellcolor{SANDgreen}\bfseries Company Name}\\
+  \hline
+  Andrey Prokopenko & prokopenkoav@ornl.gov & Oak Ridge National Laboratories\\
```
Comment (aprokop) [inline packages/muelu/doc/UsersGuide/SANDdistribution.tex]: Minor fix:
```suggestion
  Andrey Prokopenko & prokopenkoav@ornl.gov & Oak Ridge National Laboratory\\
```

Comment (lucbv) [reply packages/muelu/doc/UsersGuide/SANDdistribution.tex]: I added the change in the latest commit thanks @aprokop

Comment (jhux2) [reply packages/muelu/doc/UsersGuide/SANDdistribution.tex]: @aprokop Is your physical mailing address correct?

Comment (jhux2) [reply packages/muelu/doc/UsersGuide/SANDdistribution.tex]: @lucbv Can you update Andrey's address:
```
PO Box 2008
Bldg 5700, MS 6164
Oak Ridge, TN 37831
```

**Comment by jhux2** [comment]
@lucbv Issue #1730 may be relevant.

**Comment by lucbv** [comment]
@jhux2 I have rebase this branch and added update adresses for everyone. I have also added a small subsection: 3.3.3, that describes briefly how structured algorithms can be use and what the interface for such algorithm is.
I do not plan on adding anything mode at this point.

**Comment by lucbv** [comment]
This is the final version ready to be merged. Thanks everyone for working on this!

**Comment by jhux2** [comment]
___
*[packages/muelu/doc/UsersGuide/definitions.tex, line 66 at r1](https://reviewable.io/reviews/trilinos/trilinos/3561#-LXusI0q8DwZ6ra_cZfw:-LXusI0q8DwZ6ra_cZfx:bnlrgp3) ([raw file](https://github.com/trilinos/trilinos/blob/f961325acf2ac09059a3c007a61556ba4f98952b/packages/muelu/doc/UsersGuide/definitions.tex#L66)):*
> ```TeX
> %%% mode: latex
> %%% TeX-master: "mueluguide"
> %%% End:
> ```

I see these lines in a bunch of files, but no other changes.  Is this a side-effect of the editor you're using?
```
%%% Local Variables:
%%% mode: latex
%%% TeX-master: "mueluguide"
```

**Comment by lucbv** [comment]
@jhux2 Yes, this is set by emacs' Auctex, it basically tell pdflatex on which file to run, here: `mueluguide.tex`.

**Pull request merged**